src/components/UriViewer/LogViewer.ts

```diff
@@ -0,0 +1,224 @@
+
+import { ButtonOutline } from '../common';
+import { centeredSpinner, icon } from '../icons';
+import Modal from '../Modal';
+import { InfoBox } from '../PopupTrigger';
+/**
+ * Information needed to preview a log file.
+ * @member logUri - The URI of the log file. Must be a valid Azure blob URI. No Sas token should be appended: a fresh one will be obtained.
+ * @member logTitle - The title of the log. Displayed to the user as a tab title.
+ * @member logKey - A unique key for this particular log.
+ * @member logFilename - The filename of this particular log. Does not need to be unique.
+ */
+export type LogInfo = {
+  logUri: string;
+  logTitle: string;
+  logKey: string;
+  logFilename: string;
+};
+
+/**
+ * Props for the LogViewer component.
+ * @member modalTitle - Should represent the group of logs being displayed. E.g. "Task Logs"
+ * @member logs - An array of LogInfo objects.
+ */
+export type LogViewerProps = {
+  modalTitle: string;
+  logs: LogInfo[];
+  onDismiss: () => void;
+};
+
+/**
+ * Props for the SimpleTab component. See src/components/tabBars.js
+ */
+type SimpleTabProps = {
+  key: string;
+  title: string;
+  width: number;
+};
+
+type FetchedLogData = {
+  textContent: string | undefined; // The literal characters of the log file.
+  downloadUri: string | undefined; // The URI to use for downloading the log file. May or may not have token appended, depending on if the file is public or private.
+};
+
+const modalMaxWidth = 1100;
+const tabMaxWidth = modalMaxWidth / 4 - 20;
+
+export const LogViewer = _.flow(withDisplayName('LogViewer'))(({ modalTitle, logs, onDismiss }: LogViewerProps) => {
+  const [currentlyActiveLog, setCurrentlyActiveLog] = _.isEmpty(logs)
+    ? useState<LogInfo | undefined>(undefined)
+    : useState<LogInfo | undefined>(logs[0]);
+
+  // string = fetched log content, undefined = loading, null = error.
+  const [activeTextContent, setActiveTextContent] = useState<string | undefined | null>(undefined);
+  const [activeDownloadUri, setActiveDownloadUri] = useState<string | undefined>(undefined);
+  const signal = useCancellation();
+  const fetchLogContent = useCallback(
+    async (azureBlobUri: string): Promise<FetchedLogData | null> => {
+      if (!isAzureUri(azureBlobUri)) {
+        return null;
+      }
+      try {
+        const response = await Ajax(signal).AzureStorage.blobMetadata(azureBlobUri).getData();
+        const uri = _.isEmpty(response.azureSasStorageUrl) ? response.azureStorageUrl : response.azureSasStorageUrl;
+        return { textContent: response.textContent, downloadUri: uri };
+      } catch (e) {
+        return null;
+      }
+    },
+    [signal]
+  );
+
+  useOnMount(() => {
+    fetchLogContent(logs[0].logUri).then((content) => {
+      if (_.isEmpty(content?.textContent)) {
+        setActiveTextContent(null);
+      } else {
+        setActiveTextContent(content?.textContent);
+      }
+      setActiveDownloadUri(content?.downloadUri);
+    });
+  });
```

Comment:
It looks like the `useEffect` hook below is doing the same thing. Is this `useOnMount` necessary?